citation-categories/java/src/main/java/org/linqs/psl/examples/citationcategories/Run.java

```diff
@@ -0,0 +1,259 @@
+            inserter.loadDelimitedDataTruth(Paths.get(DATA_PATH, type, "link_obs.txt").toString());
+
+            inserter = dataStore.getInserter(model.getStandardPredicate("HasCat"), obsPartition);
+            inserter.loadDelimitedDataTruth(Paths.get(DATA_PATH, type, "hasCat_obs.txt").toString());
+
+            inserter = dataStore.getInserter(model.getStandardPredicate("HasCat"), targetsPartition);
+            inserter.loadDelimitedData(Paths.get(DATA_PATH, type, "hasCat_targets.txt").toString());
+
+            inserter = dataStore.getInserter(model.getStandardPredicate("HasCat"), truthPartition);
+            inserter.loadDelimitedDataTruth(Paths.get(DATA_PATH, type, "hasCat_truth.txt").toString());
+        }
+    }
+
+    /**
+     * Use the training data to learn weights for our rules and store them back in the model object.
+     */
+    private void learnWeights() {
+        log.info("Starting weight learning");
+
+        Partition obsPartition = dataStore.getPartition(PARTITION_LEARN_OBSERVATIONS);
+        Partition targetsPartition = dataStore.getPartition(PARTITION_LEARN_TARGETS);
+        Partition truthPartition = dataStore.getPartition(PARTITION_LEARN_TRUTH);
+
+        // This database contains all the ground atoms (targets) that we want to infer.
+        // It also includes the observed data (because we will run inference over this db).
+        Set<StandardPredicate>set = new HashSet<StandardPredicate>();
+	set.add(model.getStandardPredicate("Link"));
+	Database randomVariableDatabase = dataStore.getDatabase(targetsPartition, set, obsPartition);
+
+        // This database only contains the true ground atoms.
+        Database observedTruthDatabase = dataStore.getDatabase(truthPartition, dataStore.getRegisteredPredicates());
+
+        WeightLearningApplication wla = new MaxLikelihoodMPE(model, randomVariableDatabase, observedTruthDatabase);
+        wla.learn();
+
+        randomVariableDatabase.close();
+        observedTruthDatabase.close();
+
+        log.info("Weight learning complete");
+    }
+
+    /**
+     * Run inference to infer the unknown HasCat relationships between people.
+     */
+    private void runInference() {
+        log.info("Starting inference");
+
+        Partition obsPartition = dataStore.getPartition(PARTITION_EVAL_OBSERVATIONS);
+        Partition targetsPartition = dataStore.getPartition(PARTITION_EVAL_TARGETS);
+        Set<StandardPredicate>set = new HashSet<StandardPredicate>();
+	set.add(model.getStandardPredicate("Link"));
+	Database inferDB = dataStore.getDatabase(targetsPartition, set, obsPartition);
+
+        InferenceApplication inference = new MPEInference(model, inferDB);
+        inference.inference();
+
+        inference.close();
+        inferDB.close();
+
+        log.info("Inference complete");
+    }
+
+    /**
+     * Writes the output of the model into a file
+     */
+    private void writeOutput() {
+	    try
```

Comment:
Remove this try.
See how the simple-acquaintances does it:
https://github.com/linqs/psl-examples/blob/develop/simple-acquaintances/java/src/main/java/org/linqs/psl/examples/simpleacquaintances/Run.java#L133